libsabledb/src/commands/generic_commands.rs

```diff
@@ -196,82 +196,75 @@ impl GenericCommands {
-                            return Ok(());
-                        };
-                        if value_metadata.expiration().has_ttl() {
-                            let num = BytesMutUtils::to_u64(seconds);
-                            let mut exp = value_metadata.expiration_mut().clone();
-                            if num > exp.ttl_ms {
-                                exp.set_expire_timestamp_seconds(num)?;
-                                generic_db.put_expiration(key, &exp)?;
-                            }
+                        if num > exp.ttl_ms {
+                            exp.set_expire_timestamp_seconds(num)?;
+                            generic_db.put_expiration(key, &exp)?;
                         }
+                        builder.ok(response_buffer);
+                        return Ok(());
                     }
                     "lt" => {
                         // LT -- Set expiry only when the new expiry is less than current one
-                        let Some((_, mut value_metadata)) = generic_db.get(key)? else {
-                            return Ok(());
-                        };
-                        if value_metadata.expiration().has_ttl() {
-                            let num = BytesMutUtils::to_u64(seconds);
-                            let mut exp = value_metadata.expiration_mut().clone();
-                            if num < exp.ttl_ms {
-                                exp.set_expire_timestamp_seconds(num)?;
-                                generic_db.put_expiration(key, &exp)?;
-                            }
+                        if num < exp.ttl_ms {
+                            exp.set_expire_timestamp_seconds(num)?;
+                            generic_db.put_expiration(key, &exp)?;
                         }
+                        builder.ok(response_buffer);
+                        return Ok(());
                     }
                     _ => {
-                        // prepare error message to the user
-                        builder.error_string(response_buffer, "...");
+                        builder.error_string(response_buffer, ErrorStrings::SYNTAX_ERROR);
                         return Ok(());
                     }
                 }
             }
             (Some(key), Some(seconds), None) => {
-                if let Some(expiration) = generic_db.get_expiration(key)? {
-                    let mut exp = expiration.clone();
-                    let num = BytesMutUtils::to_u64(seconds);
-                    exp.set_expire_timestamp_seconds(num)?;
-                    generic_db.put_expiration(key, &exp)?;
-                }
+                let mut expiration = match generic_db.get_expiration(key)? {
+                    Some(expiration) => expiration,
+                    None => Expiration::default(),
+                };
+                let num = BytesMutUtils::to_u64(seconds);
```

Comment:
this needs to be `BytesMutUtils::parse::<u64>(..)` similar to what you did on line `204`